zetaclient/chains/solana/signer/signer.go

```diff
@@ -247,37 +247,45 @@ func (signer *Signer) TryProcessOutbound(
 	signer.broadcastOutbound(ctx, outbound, chainID, nonce, logger, zetacoreClient)
 }
 
-// signTx creates and signs solana tx containing provided instruction with relayer key.
+// signTx creates and signs a Solana tx containing the provided instruction with the relayer key.
+// If `alt` is non-nil and `addrs` is non-empty, the transaction will include an address lookup table.
 func (signer *Signer) signTx(
 	ctx context.Context,
 	inst *solana.GenericInstruction,
 	limit uint64,
+	alt *solana.PublicKey,
```

Comment:
Do we really need this `alt` argument as we can infer the availability of ALT from `len(addrs) > 0`?

Reply:
yea it is needed here:

```
opts = append(opts, sol.TransactionAddressTables(map[sol.PublicKey]sol.PublicKeySlice{
			*alt: addrs,
		}))
```